From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yuki Sireneva <yuki.utk8g@gmail.com>
Date: Wed, 15 Apr 2026 23:02:59 +0300
Subject: [PATCH] (Message) Show forwarded message date

This patch is adapted from Forkgram.
Original author: 23rd <23rd@vivaldi.net>

diff --git a/Telegram/SourceFiles/history/history_item_components.cpp b/Telegram/SourceFiles/history/history_item_components.cpp
index fb76cf37b4..a19969709a 100644
--- a/Telegram/SourceFiles/history/history_item_components.cpp
+++ b/Telegram/SourceFiles/history/history_item_components.cpp
@@ -11,6 +11,7 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 #include "base/qt/qt_key_modifiers.h"
 #include "base/algorithm.h"
 #include "base/options.h"
+#include "base/unixtime.h"
 #include "iv/iv_rich_page.h"
 #include "lang/lang_keys.h"
 #include "ui/effects/ripple_animation.h"
@@ -380,6 +381,9 @@ TextWithEntities HistoryMessageForwarded::phrase(
 				tr::marked);
 		}
 	}
+	if (originalDate != TimeId(0)) {
+		result.append(" " + Ui::kQBullet + u" \u2068" + Ui::FormatDateTime(base::unixtime::parse(originalDate)));
+	}
 	return result;
 }
 
